frontend/builder.go

```diff
@@ -39,6 +39,7 @@ type Compiler interface {
 	//
 	// If nbOutputs is specified, it must be >= 1 and <= f.NbOutputs
 	NewHint(f solver.Hint, nbOutputs int, inputs ...Variable) ([]Variable, error)
+	NewNamedHint(id solver.HintID, nbOutputs int, inputs ...Variable) ([]Variable, error)
```

Comment:
the name of the API confuses me a bit; 
it reads `NewNamedHint` but takes as parameter a `HintID`  

Essentially it says "give me nbOutputs variable, and tell the solver to computes them with a hint that'll be provided at solve time with id == id" right?

Reply:
> Essentially it says "give me nbOutputs variable, and tell the solver to computes them with a hint that'll be provided at solve time with id == id" right?
Exactly.

> it reads NewNamedHint but takes as parameter a HintID
You're right about that. Open to suggestions for a better name, couldn't think of one myself. Maybe `NewDeferredHint` but then again that's a bit misleading since all hints are deferred except for in test.engine

Reply:
<img width="699" alt="image" src="https://github.com/ConsenSys/gnark/assets/11349757/5bde11d0-069d-4a7a-9748-eb2b2aff5dc5">